config/initializers/protected_id.rb

```diff
@@ -13,6 +13,7 @@ module ProtectedId
   module Encoder
     def encode(id)
       return id.to_s unless PROTECT_IDS
+      return if id.nil?
```

Comment:
Should the return if `id.nil?` go first?  As written it'll always return `''` if `PROTECT_IDS` and id is nil but return nil if PROTECT_IDS is false.

Reply:
Yeah, the error itself happens in `obfuscate`. The call stack is `to_param`=>`encode`=>`obfuscate`. Sure, I can update to return a string.

Do you know why it's hitting this obfuscation code for route `hmis_admin/roles/new`? I feel like I must have done something wrong with the linking, but I can't find it.

Reply:
Actually, `encode` and `obfuscate` are only ever called as a result of `to_param`, so I'll just add the check there.